joomlarrssb.php

```diff
@@ -184,15 +184,27 @@ public function onContentAfterTitle($context, &$article, &$params, $page)
 		$siteURL = substr(JUri::root(), 0, -1);
 		$itemURL = $siteURL . JRoute::_(ContentHelperRoute::getArticleRoute($article->slug, $article->catid));
 
-		// Get the content and merge in the template; first see if $article->text is defined
-		if (!isset($article->text))
+		// Check if we have an intro text image (First priority)
+		$article->images = json_decode($article->images);
+
+		if (!empty($article->images->image_intro))
 		{
-			$article->text = $article->introtext;
+			$imageOg = $article->images->image_intro;
 		}
+		else
+		{
+			// Get the content and merge in the template; first see if $article->text is defined
+			if (!isset($article->text))
+			{
+				$article->text = $article->introtext;
+			}
 
-		// Always run this preg_match as the results are also used in the layout
-		$pattern = "/<img[^>]*src\=['\"]?(([^>]*)(jpg|gif|JPG|png|jpeg))['\"]?/";
-		preg_match($pattern, $article->text, $matches);
+			// Always run this preg_match as the results are also used in the layout
+			$pattern = "/<img[^>]*src\=['\"]?(([^>]*)(jpg|gif|JPG|png|jpeg))['\"]?/";
+			preg_match($pattern, $article->text, $matches);
+
+			$imageOg = $matches[1];
```

Comment:
This should check if `$matches[1]` is defined also as it wouldn't be if the article text doesn't contain any matches.